clap_derive/src/dummies.rs

```diff
@@ -76,11 +76,13 @@ pub fn args(name: &Ident) {
 pub fn arg_enum(name: &Ident) {
     append_dummy(quote! {
         impl clap::ArgEnum for #name {
-            const VARIANTS: &'static [&'static str] = &[];
+            fn value_variants<'a>() -> &'a [Self]{
+                unimplemented!()
+            }
             fn from_str(_input: &str, _case_insensitive: bool) -> Result<Self, String> {
                 unimplemented!()
             }
-            fn as_arg(&self) -> Option<&'static str> {
+            fn arg_value<'a>(&self) -> Option<clap::ArgValue<'a>>{
```

Comment:
WDYT about naming this `as_arg_value`?

Reply:
The call might allocate and so `as_` wouldn't work, though `to_` would.  I can go either way.

Reply:
I renamed it to `to_arg_value` now